sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -157,7 +188,14 @@ public static ClientPipeline Create(
         int beforeTransportIndex = index;
 
         // Add the transport.
-        policies[index++] = options.Transport ?? HttpClientPipelineTransport.Shared;
+        if (loggingOptions == null || loggingOptions.ShouldUseDefaultPipelineTransport())
```

Comment:
Why does it make sense to ask the logging options which transport to add to the pipeline?